Test/Helper/StoreTest.php

```diff
@@ -0,0 +1,134 @@
+    {
+        parent::__construct();
+        $objectManager = new ObjectManager($this);
+        /** @var Context */
+        $context = $this->createMock(Context::class);
+
+        $store1 = [
+            'name'  => 'store1',
+            'code'  => 's1',
+            'id'    => 1,
+            'url'   => 'store1.com',
+            'active' => true
+        ];
+
+        $store2 = [
+            'name'  => 'store2',
+            'code'  => 's2',
+            'id'    => 2,
+            'url'   => 'store2.com',
+            'active' => true
+        ];
+
+        $store1Mock = $this->getMockBuilder(Store::class)
+            ->setMethods(['getName', 'getCode', 'getStoreId', 'getBaseUrl', 'isActive',])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $store1Mock->method('getName')->willReturn($store1['name']);
+        $store1Mock->method('getCode')->willReturn($store1['code']);
+        $store1Mock->method('getStoreId')->willReturn($store1['id']);
+        $store1Mock->method('getBaseUrl')->willReturn($store1['url']);
+        $store1Mock->method('isActive')->willReturn($store1['active']);
+
+        $store2Mock = $this->getMockBuilder(Store::class)
+            ->setMethods(['getName', 'getCode', 'getStoreId', 'getBaseUrl', 'isActive'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $store2Mock->method('getName')->willReturn($store2['name']);
+        $store2Mock->method('getCode')->willReturn($store2['code']);
+        $store2Mock->method('getStoreId')->willReturn($store2['id']);
+        $store2Mock->method('getBaseUrl')->willReturn($store2['url']);
+        $store2Mock->method('isActive')->willReturn($store2['active']);
+
+        $storeCollectionMock = $objectManager->getCollectionMock(Collection::class, [$store1Mock, $store2Mock]);
+        $storeCollectionMock->method('addWebsiteFilter')->willReturn([$store1Mock]);
+
+        $storeMock = $this->getMockBuilder(Store::class)
+        ->setMethods(['getCollection'])
+        ->disableOriginalConstructor()
+        ->getMock();
+        $storeMock->method('getCollection')->willReturn($storeCollectionMock);
+
+        $storeManagerTemp = $this->createMock(StoreManagerInterface::class);
+        $storeManagerTemp->method('getStores')->willReturn($storeCollectionMock);
+        /** @var StoreManagerInterface */
+        $storeManager = $storeManagerTemp;
+
+        $storeFactoryTemp = $this->createMock(StoreFactory::class);
+        $storeFactoryTemp->method('create')->willReturn($storeMock);
+        $storeFactory = $storeFactoryTemp;
+
+        $this->store1 = $store1;
+        $this->store2 = $store2;
+
+        $this->store = new StoreHelper(
+            $context,
+            $storeFactory,
+            $storeManager
+        );
+    }
+    /**
+     *  Store::getUserStores() should return all stores in the collection
+     */
+    public function testGetUserStores(): void
+    {
+        $this->assertEquals(
```

Comment:
I think it would be better to put the actual tests in these test methods, rather than just the assertions. For any common code that needs to execute before/after each test, you could create `setUp()` and `tearDown()` methods.

Going forward, we'll probably want/need the extra flexibility of not having to execute all of the actual tests in the constructor.

Reply:
it's not actually executing the tests in the constructor. Unfortunately the extensive amount of mocking is just to replicate functionality of magento's built in store collections (it utilizes a number of magic methods which don't work in stubs), which we we don't want to test. The tests themselves really only deal with the the store helper retrieving the correct slices of the store collections.

Reply:
I theory I could write the tests as : 
```php
    public function testGetAllStores(): void
    {
        $actual = $this->store->getAllStores();
        $expected = [
                $this->store1, $this->store2
            ];
        $this->assertEquals(
            $expected,
            $actual
        );
    }
```
but the alternative is a good bit more terse without being that much less readable

Reply:
I see. Would it be possible to set up those mocks in `setUp()` (so they can be torn down afterward in `tearDown()`) and/or to do @aldahick's suggestion of moving all that code out to a helper?

Reply:
`setUp` seems like a good solution potentially here if we want to maintain test flexibility. I like @aldahick's suggestion regarding a helper class for Tests but want to propose if that should be it's own ticket where we aim to make a class that provides flexibility and customization in generating test/mock objects.

Reply:
yeah, I can definitely move it over to setUp()

Reply:
Thanks for doing this. Would you mind also creating a corresponding `tearDown()` method to clean up the mocks? The idea is that `setUp()` runs before each test, and `tearDown()` runs after each test, so that it never matters what order the tests run in.

Reply:
done, though even without the teardown it didn't matter what order the tests were run in, since all the tests did is fetch different slices of the same data from the collection